### PR TITLE
Fix overrides to be provider-specific so rsync is only used for AWS and cache Cloudera Hadoop downloads with Virtualbox.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -147,7 +148,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     end
 
     # Exclude some directories that can grow very large from syncing
-    config.vm.synced_folder ".", "/vagrant", type: "rsync", :rsync_excludes => ['.git', 'core/data/', 'logs/', 'system_test/']
+    override.vm.synced_folder ".", "/vagrant", type: "rsync", :rsync_excludes => ['.git', 'core/data/', 'logs/', 'system_test/']
```
Comment (granders) [inline Vagrantfile]: Looks like we never updated these rsync_excludes to make sense for muckrake. 

E.g. ['.git/', '.vagrant'/, 'results/']

File: vagrant/base.sh [hunk 2]
```diff
@@ -60,14 +60,18 @@ fi
 chmod a+rwx /mnt
 
 # Install and configure CDH
-pushd /opt/
 if [ -h /opt/hadoop-cdh ]; then
     # reset symlink
     rm -f /opt/hadoop-cdh
 fi
-if [ ! -e /opt/hadoop-2.5.0-cdh5.3.0 ]; then
+pushd /tmp/vagrant-downloads
```
Comment (granders) [inline vagrant/base.sh]: on aws this caused an error (/tmp/vagrant-downloads doesn't exist)

**Comment by ewencp** [comment]
@granders At long last, I finally updated this and tested on aws. The only open question is what directories should be in the excludes. I think the set I included are reasonable, but let me know if I missed anything else that might have large quantities of data.

**Comment by granders** [comment]
thanks! LGTM

**Pull request merged**